src/javascript/app_2/pages/trading/components/elements/tabs.jsx

```diff
@@ -0,0 +1,128 @@
+                    </React.Fragment>
+                ))}
+                </TabsWrapper>
+                <p className='tab-content'>{this.props.list[this.state.active].content}</p>
+            </div>
+        );
+    }
+};
+
+class TabsWrapper extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.state = {
+            sizes: {},
+        };
+        this.els = {};
+    }
+
+    componentDidMount() {
+        this.getSizes();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.children !== this.props.children && prevProps.active !== this.props.active) {
+            this.getSizes();
+        }
+    }
+
+    getSizes() {
+        const rootBounds = this.root.getBoundingClientRect();
+
+        const sizes = {};
+
+        Object.keys(this.els).forEach((key) => {
+            const el = this.els[key];
+            const bounds = el.getBoundingClientRect();
+
+            const left = bounds.left - rootBounds.left;
+            const right = rootBounds.right - bounds.right;
+
+            sizes[key] = {left, right};
+        });
+
+        this.setState({sizes});
+        return sizes;
+    }
+
+    render() {
+        // console.log(JSON.stringify(this.state, null, 2));
+        return (
+          <div
+            className='tab-wrapper'
+            ref={el => this.root = el}
+          >
+            {React.Children.map(this.props.children, (child, i) => {
+                const className = 'tab';
+                return (
+                  <React.Fragment key={i}>
+                      <div
+                        className={child.key === this.props.active ? `${className} tab--active` : className}
+                        onClick={() => {
+                            this.props.onChange(child.key);
+                        }}
+                        ref={el => this.els[child.key] = el}
+                      >
+                        {child}
+                      </div>
+                  </React.Fragment>
+                );
+            })}
+            <div
+              className='tab_underline'
+              style={this.getUnderlineStyle()}
+            />
+          </div>
+        );
+    }
+
+    getUnderlineStyle() {
+        if (this.props.active == null || Object.keys(this.state.sizes).length === 0) {
+            return {left: '0', right: '100%'};
+        }
+
+        const size = this.state.sizes[this.props.active];
+
+        return {
+            left      : `${size.left}px`,
+            right     : `${size.right}px`,
+            transition: 'left 0.2s, right 0.25s',
+        };
+    }
+}
+
+// Dummy values
+const content = {
+    1: { header: 'Tab 1', content: 'Content 1' },
+    2: { header: 'Tab 2', content: 'Content 2' },
+    3: { header: 'Tab 3', content: 'Content 3' },
+    4: { header: 'Tab 4', content: 'Content 4' },
+};
+
+Tabs.defaultProps = {
+    alignment: 'left',
+    list     : content,
+};
```

Comment:
nitpick: we need to comment out dummy default props, otherwise they will add to bundle size.

Reply:
We'll remove all the defaultProps eventually, but currently they're mostly there as a reference point for rendering the components. I'll note to remove this eventually, thanks.

Reply:
👍